src/Workspaces/Core/Portable/Shared/Collections/IntervalTree`1.cs

```diff
@@ -309,17 +313,51 @@ public IEnumerator<T> GetEnumerator()
                 yield break;
             }
 
+            // The bool indicates if this is the first time we are seeing the node.
+            var candidates = new Stack<ValueTuple<Node,bool>>();
+            candidates.Push(ValueTuple.Create(root, true));
+            while (candidates.Count != 0)
+            {
+                var currentTuple = candidates.Pop();
+                var currentNode = currentTuple.Item1;
+                if (currentNode != null)
+                {
+                    if (currentTuple.Item2)
+                    {
+                        // First time seeing this node.  Mark that we've been seen and recurse
+                        // down the left side.  The next time we see this node we'll yield it
+                        // out.
+                        candidates.Push(ValueTuple.Create(currentNode.Right, true));
+                        candidates.Push(ValueTuple.Create(currentNode, false));
+                        candidates.Push(ValueTuple.Create(currentNode.Left, true));
+                    }
+                    else
+                    {
+                        yield return currentNode.Value;
+                    }
+                }
+            }
+        }
+
+        // For testing purposes only.  Allows a test to verify the actual AVL structure of the tree.
```

Comment:
I was struggling with that myself.  Frankly, neither ordering makes testing work out.  Take an in-order walk.  An in-order walk will produce the same result for:

```
A
 \
  B
   \
    C
     \
      D
```

As it would for

```
   B
  / \
A    C
      \
       D
```

And yet one of these is balanced properly, and the other is not.

IMO, to actually test the rotations we'd need tests that validated the internal node-structure.  And, if we don't want to expose that, we should probably delete the tests and just test the exposed behavior.